scripts/asm-installer/install_asm

```diff
@@ -1524,6 +1553,10 @@ configure_package() {
     run kpt cfg set asm anthos.servicemesh.hub "${_CI_ASM_IMAGE_LOCATION}"
     run kpt cfg set asm anthos.servicemesh.tag "${RELEASE}"
   fi
+  if [[ "${ENABLE_HUB_IDNS}" -eq 1 ]]; then
+    run kpt cfg set asm gcloud.project.environProjectID "${ENVIRON_PROJECT_ID}"
+    run kpt cfg set asm anthos.servicemesh.hubMembershipID "${HUB_MEMBERSHIP_ID}"
+  fi
```

Comment:
I'm not sure if it matters for Hub IDNS specifically, but if you call the script with --print-config then `ENVIRON_PROJECT_ID` and `HUB_MEMBERSHIP_ID` might not be set. If that matters then you should probably add a guard to set them, something like

```
if [[ -z ENVIRON || -z HUB_ID ]]; then
  get_hub_workload_identity > /dev/null
fi
```